### ISSUE TITLE
Cannot set LineEdit input-type property

### ISSUE THREAD

**Issue by kivimango** [slint-ui/slint#2408]
Hello ! Im having issues setting LineEdit's input type property both in the online editor and/or in VsCode.

Example 1:

When using named LineEdit, it fails to compile with the error message:
`inmemory://model/2:10:inmemory://model/2:10: Cannot take reference of an element`

```slint
import { LineEdit } from "std-widgets.slint";
export component Example inherits Window {
    width: 200px;
    height: 25px;
    password := LineEdit {
    border-color: gray;
        font-size: 14px;
        width: parent.width;
        height: parent.height;
        input-type: password;
        placeholder-text: "Enter text here";
    }
}
```

vs when im using an unnamed LineEdit:
```slint
import { LineEdit } from "std-widgets.slint";
export component Example inherits Window {
    width: 200px;
    height: 25px;
    LineEdit {
    border-color: gray;
        font-size: 14px;
        width: parent.width;
        height: parent.height;
        input-type: password;
        placeholder-text: "Enter text here";
    }
}
```
its completely okay in the online editor.
But in VSCode this is still refuses to compile with :
```slint
import { LineEdit } from "std-widgets.slint";
export component Example inherits Window {
    width: 200px;
    height: 25px;
    LineEdit {
    border-color: gray;
        font-size: 14px;
        width: parent.width;
        height: parent.height;
        input-type: text;
        placeholder-text: "Enter text here";
    }
}
```
`Cannot convert string to enum InputType`
when the input-type property is set to `text` explicitly.

Also i can't set the border related properties in VSCode, however the online editor accepts them out of the box.
`border-color`
`border-width`
`border-radius`
`Unknown property border-width in LineEdit`

**Comment by ogoffart** [comment]
The problem here is the lookup rules
the id always come first. So if you type `password`, it thinks you mean the id. Enum values always come last. The Fix here is to qualify them:  `InputType.password`

We could see that since there is no `.` this is probably not an id and continue the lookup, but that'd be a bit magic.

> when the input-type property is set to text explicitly.

It seems like it thinks you are referring to the `text` property. This i cannot reproduce. (but this would have been broken using the old syntax that had different lookup rules

> Also i can't set the border related properties in VSCode, however the online editor accepts them out of the box.

This was caused with extra properties being exposed by mistake depending of the style. This is fixed in 1.0 (by harmonizing the styles and removing these properties)

**Comment by kivimango** [comment]
Thank you.
I just realized that im using the enum variant 'password' as an identifiter.
This edge case should be documented i think.

For the border-color property, i hoped you didnt changed the visibility to private meanwhile.
I moved on by reimplementing my own LineEdit.
